File: samples/java/SupportChat/Server/Support/src/main/java/systems/zlink/samples/supportchat/server/support/spots/conversationspot/SetTypingHandler.java
package systems.zlink.samples.supportchat.server.support.spots.conversationspot;

import systems.zlink.framework.ZLinkMessageContext;
import systems.zlink.framework.spots.ZLinkSpotActorSendHandler;
import systems.zlink.samples.supportchat.server.support.actors.SupportUserActor;
import systems.zlink.samples.supportchat.shared.contracts.Messages;

import java.util.concurrent.CompletableFuture;
import java.util.concurrent.CompletionStage;

public final class SetTypingHandler
        implements ZLinkSpotActorSendHandler<
                ConversationSpot, SupportUserActor, Messages.SetTypingMsg> {
    @Override
    public CompletionStage<Void> handle(
            ConversationSpot spot,
            SupportUserActor actor,
            ZLinkMessageContext context,
            Messages.SetTypingMsg message) {
        spot.setTyping(actor, message);
        return CompletableFuture.completedFuture(null);
    }
}

File: samples/kotlin/SupportChat/Server/Support/src/main/kotlin/systems/zlink/samples/kotlin/supportchat/server/support/infrastructure/zlink/spots/conversationspot/handlers/SetTypingHandler.kt
package systems.zlink.samples.kotlin.supportchat.server.support.infrastructure.zlink.spots.conversationspot.handlers

import systems.zlink.framework.ZLinkMessageContext
import systems.zlink.framework.kotlin.ZLinkSuspendingSpotActorSendHandler
import systems.zlink.samples.kotlin.supportchat.server.support.infrastructure.zlink.actors.SupportUserActor
import systems.zlink.samples.kotlin.supportchat.server.support.infrastructure.zlink.spots.conversationspot.ConversationSpot
import systems.zlink.samples.kotlin.supportchat.shared.contracts.SetTypingMsg

class SetTypingHandler :
    ZLinkSuspendingSpotActorSendHandler<ConversationSpot, SupportUserActor, SetTypingMsg> {
    override suspend fun handle(
        spot: ConversationSpot,
        actor: SupportUserActor,
        context: ZLinkMessageContext,
        message: SetTypingMsg,
    ) {
        spot.setTyping(actor, message)
    }
}
